woocommerce/woocommerce
# Migrate `@woocommerce/data` user and use-select-with-refresh to TS

### All Submissions:






### Changes proposed in this Pull Request:



Part of https://github.com/woocommerce/woocommerce/issues/33105

This PR migrates `./user` and `use-select-with-refresh.js` to TS

### How to test the changes in this Pull Request:

1. Should run `pnpm -- turbo run build --filter=@woocommerce/data` without errors
2. Review changes



### Other information:




### FOR PR REVIEWER ONLY:
